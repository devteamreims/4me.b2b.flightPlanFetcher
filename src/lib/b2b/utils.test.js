import {
  flightSoapEnvelope,
  airspaceServiceSoapEnvelope,
} from './utils';

describe('b2b envelopes', () => {
  describe('flightService', () => {
    test('should check arguments', () => {
      expect(() => flightSoapEnvelope()).toThrowErrorMatchingSnapshot();
      expect(() => flightSoapEnvelope({id: 1})).toThrowErrorMatchingSnapshot();
    });

    test('should match snapshot', () => {
      expect(flightSoapEnvelope('payload')).toMatchSnapshot();
    });
  });

  describe('airspaceService', () => {
    test('should check arguments', () => {
      expect(() => airspaceServiceSoapEnvelope()).toThrowErrorMatchingSnapshot();
      expect(() => airspaceServiceSoapEnvelope({id: 1})).toThrowErrorMatchingSnapshot();
    });

    test('should match snapshot', () => {
      expect(airspaceServiceSoapEnvelope('payload')).toMatchSnapshot();
    });
  });
});
