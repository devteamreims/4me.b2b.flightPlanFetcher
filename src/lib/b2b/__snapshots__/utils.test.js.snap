exports[`b2b envelopes airspaceService should check arguments 1`] = `"Invalid argument : please provide a String"`;

exports[`b2b envelopes airspaceService should check arguments 2`] = `"Invalid argument : please provide a String"`;

exports[`b2b envelopes airspaceService should match snapshot 1`] = `
"
    <soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:air=\"eurocontrol/cfmu/b2b/AirspaceServices\">
      <soapenv:Header/>
      <soapenv:Body>
        payload
      </soapenv:Body>
    </soapenv:Envelope>
  "
`;

exports[`b2b envelopes flightService should check arguments 1`] = `"Invalid argument : please provide a String"`;

exports[`b2b envelopes flightService should check arguments 2`] = `"Invalid argument : please provide a String"`;

exports[`b2b envelopes flightService should match snapshot 1`] = `
"
    <soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:flig=\"eurocontrol/cfmu/b2b/FlightServices\">
      <soapenv:Header/>
      <soapenv:Body>
        payload
      </soapenv:Body>
    </soapenv:Envelope>
  "
`;
